pkg/platform/runtime/setup/buildlog/artifactlogs.go

```diff
@@ -0,0 +1,70 @@
+package buildlog
+
+import (
+	"context"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/api/buildlogstream"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+	"github.com/gorilla/websocket"
+)
+
+type logConnection struct {
+	conn *websocket.Conn
+	log  *ArtifactLog
+}
+
+// ArtifactLogs manages websocket connections to the build-log-streamer for artifact specific logs
+// Unfortunately, we need to spawn a new connection for every artifactID that prints its loglines in real-time. If artifact logs for builds that happened in the past are requested, the information can be streamed in a single connection.
```

Comment:
During my tests this morning, I realized that I had to add a separate connection for each and every artifactID that I want to receive real-time build log information from.  Fortunately, this is only relevant in the `ACTIVESTATE_BUILD_VERBOSE=true` case right now.